yodeploy/hooks/daemon.py

```diff
@@ -0,0 +1,151 @@
+            self.app.template(template, target)
+            self.reload()
+            self.restart(name)
+            self.installed_daemons = True
+
+    def iter_templates(self):
+        if not os.path.exists(self.template_dir):
+            return
+        for fn in os.listdir(self.template_dir):
+            if not fn.endswith('.template'):
+                continue
+            name = fn.split('.')[0]
+            template = os.path.join(self.name, fn)
+            target = os.path.join(self.target_path, fn.rsplit('.', 1)[0])
+            yield name, template, target
+
+    def reload(self):
+        raise NotImplemented
+
+    def restart(self, task):
+        raise NotImplemented
+
+    @property
+    def template_dir(self):
+        return self.app.deploy_path('deploy', 'templates', self.name)
+
+
+class Supervisor(Manager):
+    name = 'supervisord'
+    target_path = '/etc/supervisor/conf.d'
+
+    def is_available(self):
+        return os.path.exists('/usr/bin/supervisord')
+
+    def reload(self):
+        try:
+            subprocess.check_call(('supervisorctl', 'reread'))
+            subprocess.check_call(('supervisorctl', 'update'))
+        except subprocess.CalledProcessError:
+            log.error('Unable to update supervisord configs')
+
+    def restart(self, task):
+        try:
+            subprocess.call(('supervisorctl', 'stop', task))
+            subprocess.check_call(('supervisorctl', 'start', task))
+        except subprocess.CalledProcessError:
+            log.error('Unable to restart %s supervisord task', task)
+
+
+class SystemD(Manager):
+    name = 'systemd'
+    target_path = '/etc/systemd/system'
+
+    def is_available(self):
+        return os.path.exists('/bin/systemctl')
+
+    def reload(self):
+        try:
+            subprocess.check_call(('systemctl', 'daemon-reload'))
+        except subprocess.CalledProcessError:
+            log.error('Unable to reload systemd config')
+
+    def restart(self, task):
+        try:
+            subprocess.check_call(('systemctl', 'enable', task))
+        except subprocess.CalledProcessError:
+            log.error('Unable to install %s systemd task', task)
+        try:
+            subprocess.check_call(('systemctl', 'restart', task))
+        except subprocess.CalledProcessError:
+            log.error('Unable to restart %s systemd task', task)
+
+
+class Upstart(Manager):
+    name = 'upstart'
+    target_path = '/etc/init'
+
+    def is_available(self):
+        return os.path.exists('/sbin/initctl')
+
+    def reload(self):
+        pass
+
+    def restart(self, task):
+        try:
+            subprocess.call(('service', task, 'stop'))
+            subprocess.check_call(('service', task, 'start'))
+        except subprocess.CalledProcessError:
+            log.error('Unable to restart %s upstart task', task)
+
+
+class DaemonApp(TemplatedApp):
+    def __init__(self, *args, **kwargs):
+        super(TemplatedApp, self).__init__(*args, **kwargs)
```

Comment:
This init is weird.

Is using `TemplatedApp` instead of `DaemonApp` a typo? If yes, this init seems like can be purged. If no, why skip the init of `TemplatedApp`?